tests/components/modbus/test_modbus_sensor.py

```diff
@@ -546,6 +547,36 @@ async def test_all_sensor(hass, cfg, regs, expected):
             [0x0101],
             "257",
         ),
+        (
+            {
+                CONF_COUNT: 1,
+                CONF_REGISTER_SIZE: 4,
+                CONF_DATA_TYPE: DATA_TYPE_CUSTOM,
+                CONF_STRUCTURE: ">L",
+            },
+            [0x11223344],
+            "287454020",
+        ),
+        (
+            {
+                CONF_COUNT: 2,
+                CONF_REGISTER_SIZE: 2,
+                CONF_DATA_TYPE: DATA_TYPE_CUSTOM,
+                CONF_STRUCTURE: ">L",
+            },
+            [0x1122, 0x3344],
+            "287454020",
+        ),
+        (
+            {
+                CONF_COUNT: 1,
+                CONF_REGISTER_SIZE: 5,
+                CONF_DATA_TYPE: DATA_TYPE_CUSTOM,
+                CONF_STRUCTURE: ">5B",
+            },
+            [0x0102030405],
+            "1,2,3,4,5",
+        ),
```

Comment:
The swap is sensitive to the register_size, so have a couple of examples, is needed.

Reply:
Maybe the only test needed is 
"count 1, size 4" vs "count 2, size 2"

All the rest is default